src/commonControls/numberControl/NumberControl.ts

```diff
@@ -594,6 +670,33 @@ export class NumberControl extends Control implements InteractionModelContributo
         }
     }
 
+    private isScreenEvent(input: ControlInput) {
+        try {
+            okIf(InputUtil.isAPLUserEventWithMatchingControlId(input, this.id));
+            this.handleFunc = this.handleScreenEvent;
+            return true;
+        } catch (e) {
+            return falseIfGuardFailed(e);
+        }
+    }
+
+    private handleScreenEvent(input: ControlInput, resultBuilder: ControlResultBuilder): void {
+        const value = (input.request as interfaces.alexa.presentation.apl.UserEvent).arguments![1];
```

Comment:
minor.  we can now use the assert() function to test things and inform the compiler about things which avoids the need to use ! (which is a less obvious assert)
see AssertinUtils.assert.  I'm not sure if it is convenient / worthwhile here, but I'm trying to use it more.

see if this works..note there is no ! on second line.
```
assert((input.request as interfaces.alexa.presentation.apl.UserEvent).arguments !== undefined);
const value = (input.request as interfaces.alexa.presentation.apl.UserEvent).arguments[1];
```

Reply:
To make this work we need something like the following to make it more explicit for the compiler:
```
const userEventArguments = (input.request as interfaces.alexa.presentation.apl.UserEvent).arguments;
assert(userEventArguments !== undefined);
const value = userEventArguments[1];
```
In this case I don't think it's convenient since we just need to extract the value and don't need the arguments for anything else (unless we want to move completely away from ! everywhere).